### PR TITLE
📖 [ION-283] document langchain tool

### PR DESCRIPTION
**Pull request by gmkohler**
Not sure what should live in here versus mintlify.

Any and all suggestions are welcome for improving the docs.

I broke out "publish" into a separate ticket ION-285.  If we make the repo private after publishing to PyPI then I'd imagine we should move all of this to Mintlify and change these docs to focus on developing it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,74 @@
 # Ionic Langchain
 
-Pre-release. Do not use this.
+Ionic Langchain provides a wrapper around the Ionic Commerce's SDK for use as a `Tool` in a custom Langchain agent.  This tool will enable e-commerce for your agent, allowing your users to ask for product recommendations and purchase products through the agent chat interface.
```
Comment (gmkohler) [inline README.md:3]: @jdmccarty I opted to write my own copy here but please suggest improvements.  As a developer the website copy doesn't tell me a lot in plain english about what the tool is doing.

File: README.md [hunk 2]
```diff
@@ -1,15 +1,74 @@
...
+or `poetry`:
+
+```sh
+poetry add git+https://github.com/ioniccommerce/ionic_langchain.git#v0.1.2
+```
 
 ## Usage
 
-Pre-release. Do not use this.
+```python
```
Comment (gmkohler) [inline README.md:23]: more or less adapting what we have written in our demobot; happy to make modifications.

File: README.md [hunk 3]
```diff
@@ -1,15 +1,74 @@
 # Ionic Langchain
 
-Pre-release. Do not use this.
+Ionic Langchain provides a wrapper around the Ionic Commerce's SDK for use as a `Tool` in a custom Langchain agent.  This tool will enable e-commerce for your agent, allowing your users to ask for product recommendations and purchase products through the agent chat interface.
 
 ## Installation
 
-Pre-release. Do not use this.
+This tool requires at least `langchain@0.0.350` and can work with any greater patch release the `0.0.x` series.
```
Comment (gmkohler) [inline README.md:7]: I've never written a public package; not sure about python version compatibility or how we should be managing that.  I figure we don't need that right away but IMO worth figuring out.

File: pyproject.toml [hunk 4]
```diff
@@ -8,8 +8,8 @@ packages = [{include = "ionic_langchain"}]
 
 [tool.poetry.dependencies]
```
Comment (gmkohler) [inline pyproject.toml:9]: Do we want to add more metadata here in the effort towards publishing on PyPI?

https://packaging.python.org/en/latest/tutorials/packaging-projects/#configuring-metadata

File: README.md [hunk 5]
```diff
@@ -1,15 +1,74 @@
...
+
+You can install the package from GitHub using `pip`:
+
+```sh
+pip install git+https://github.com/ioniccommerce/ionic_langchain.git#v0.1.2
```
Comment (gmkohler) [inline README.md:12]: The official python packaging docs use `python3 -m pip` so maybe that's better here?

```suggestion
python3 -m pip install git+https://github.com/ioniccommerce/ionic_langchain.git#v0.1.2
```

File: pyproject.toml (hunk 4 above)
Comment (stewartjarod) [reply pyproject.toml:9]: Eventually 💯

**Review by stewartjarod — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (stewartjarod) [reply README.md:7]: We should only support the latest LTS for the start. 

When someone pays us money and needs a specific version that's another story imo.
